newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/CollisionMaskHelper.js

```diff
@@ -1,147 +1,154 @@
-    }
-    if (foundNonTransparent) break;
-  }
-  if (minX === null) return null;
-
-  // Step 4: Scanning Columns from Right to Left, until we find a non-transparent pixel
-  for (let x = width - 1; x >= 0; x--) {
-    let foundNonTransparent = false;
-    for (let y = minY; y <= maxY; y++) {
-      const idx = (y * width + x) << 2; // Calculate the pixel index
-
-      const alpha = jimpImage.bitmap.data[idx + 3];
-      if (alpha >= PIXEL_TRANSPARENCY_THRESHOLD) {
-        maxX = x;
-        foundNonTransparent = true;
-        break;
+      if (minY === null) {
+        reject(new Error('No non-transparent pixel found.'));
+        return;
       }
-    }
-    if (foundNonTransparent) break;
-  }
-  if (maxX === null) return null;
-
-  const collisionMaskWidth = maxX - minX + 1;
-  const collisionMaskHeight = maxY - minY + 1;
-  const collisionMaskXCenter = (minX + maxX + 1) / 2;
-  const collisionMaskYCenter = (minY + maxY + 1) / 2;
-  if (collisionMaskWidth <= 0 || collisionMaskHeight <= 0) return null;
-
-  const newPolygon = gd.Polygon2d.createRectangle(
-    collisionMaskWidth,
-    collisionMaskHeight
-  );
-  newPolygon.move(collisionMaskXCenter, collisionMaskYCenter);
-  const polygons = new gd.VectorPolygon2d();
-  polygons.push_back(newPolygon);
-
-  return polygons;
+
+      // Step 2: Scanning Rows from Bottom to Top, until we find a non-transparent pixel
+      for (let y = height - 1; y >= 0; y--) {
+        let foundNonTransparent = false;
+        for (let x = 0; x < width; x++) {
+          const idx = (y * width + x) << 2; // Calculate the pixel index
+
+          const alpha = pixels[idx + 3];
+          if (alpha >= PIXEL_TRANSPARENCY_THRESHOLD) {
+            maxY = y;
+            foundNonTransparent = true;
+            break;
+          }
+        }
+        if (foundNonTransparent) break;
+      }
+      if (maxY === null) {
+        reject(new Error('No non-transparent pixel found.'));
+        return;
+      }
+
+      // Step 3: Scanning Columns from Left to Right, until we find a non-transparent pixel
+      for (let x = 0; x < width; x++) {
+        let foundNonTransparent = false;
+        for (let y = minY; y <= maxY; y++) {
+          const idx = (y * width + x) << 2; // Calculate the pixel index
+
+          const alpha = pixels[idx + 3];
+          if (alpha >= PIXEL_TRANSPARENCY_THRESHOLD) {
+            minX = x;
+            foundNonTransparent = true;
+            break;
+          }
+        }
+        if (foundNonTransparent) break;
+      }
+      if (minX === null) {
+        reject(new Error('No non-transparent pixel found.'));
+        return;
+      }
+
+      // Step 4: Scanning Columns from Right to Left, until we find a non-transparent pixel
+      for (let x = width - 1; x >= 0; x--) {
+        let foundNonTransparent = false;
+        for (let y = minY; y <= maxY; y++) {
+          const idx = (y * width + x) << 2; // Calculate the pixel index
+
+          const alpha = pixels[idx + 3];
```

Comment:
Could you define a function that would be `isPixelAboveTransparencyThreshold(x, y) ?`. I think it could help read this big function.